### DISCUSSION TITLE
scan all function need to chain exec() call ?

### DISCUSSION THREAD

**Discussion by lastmirage** [dynamoose/dynamoose#1431] [Q&A]
In doc,

model.scan().all() will return all items in table.

But it just returns ItemRetriever and I need to call exec() to get result items

Which is it correct?

And if I use model.scan().all().exec(), it will return type of any.
But if I use model.scan().exec(), it will return type of ScanResponse<Item> as expected

I think all().exec() also return ScanResponse<Item> type.

Is it a bug? or do I need to casting it for use ?

**Comment by halan** [comment]
Maybe related to https://github.com/dynamoose/dynamoose/pull/1488
